Auth0/AuthProvider.swift

```diff
@@ -35,8 +35,27 @@ import UIKit
          ...
          return transaction
      }
+
  ```
  */
 public protocol AuthProvider {
     func login(withConnection connection: String, scope: String, parameters: [String: Any]) -> NativeAuthTransaction
+
+    /**
+     Some IdP authenticators like Auth0Twitter rely upon accessibility of on device social accounts.
+     If these are not accessible/available this pre-login check lets you know you should provide
+     alternative authentication means such as Auth0.WebAuth.
+     
+     - returns: Bool
+     */
+    func canUseNative() -> Bool
```

Comment:
Thx, amusingly I did call it `isAvailable()` originally then got conflicted looking at naming in v1 twitter.  Should have stuck with original name 👍